packages/web-app-files/src/helpers/resource/conflict-handling/conflictDialog.ts

```diff
@@ -0,0 +1,176 @@
+    const resolvedConflicts = []
+    for (const conflict of allConflicts) {
+      // Resolve conflicts accordingly
+      if (doForAllConflicts) {
+        conflict.strategy = doForAllConflictsStrategy
+        resolvedConflicts.push(conflict)
+        continue
+      }
+
+      // Resolve next conflict
+      const conflictsLeft = allConflicts.length - count
+      const isSingleConflict = conflictsLeft === 1
+      const result: ResolveConflict = await this.resolveFileExists(
+        conflict.resource,
+        conflictsLeft,
+        isSingleConflict
+      )
+      conflict.strategy = result.strategy
+      resolvedConflicts.push(conflict)
+      count += 1
+
+      // User checked 'do for all x conflicts'
+      if (!result.doForAllConflicts) continue
+      doForAllConflicts = true
+      doForAllConflictsStrategy = result.strategy
+    }
+    return resolvedConflicts
+  }
+
+  resolveFileExists(
+    resource: Resource,
+    conflictCount: number,
+    isSingleConflict: boolean,
+    suggestMerge = false
+  ): Promise<ResolveConflict> {
+    return new Promise<ResolveConflict>((resolve) => {
+      let doForAllConflicts = false
+      const modal = {
+        variation: 'danger',
+        icon: 'alarm-warning',
+        title: this.$gettext('File already exists'),
+        message: this.$gettextInterpolate(
+          resource.isFolder
+            ? this.$gettext('Folder with name %{name} already exists.')
+            : this.$gettext('File with name %{name} already exists.'),
+          { name: resource.name },
+          true
+        ),
+        cancelText: this.$gettext('Skip'),
+        confirmText: this.$gettext('Keep both'),
+        buttonSecondaryText: suggestMerge ? this.$gettext('Merge') : this.$gettext('Replace'),
+        checkboxLabel: isSingleConflict
+          ? ''
+          : this.$gettextInterpolate(
+              this.$gettext('Apply to all %{count} conflicts'),
+              { count: conflictCount },
+              true
+            ),
+        onCheckboxValueChanged: (value) => {
+          doForAllConflicts = value
+        },
+        onCancel: () => {
+          this.hideModal()
+          resolve({ strategy: ResolveStrategy.SKIP, doForAllConflicts } as ResolveConflict)
+        },
+        onConfirmSecondary: () => {
+          this.hideModal()
+          const strategy = suggestMerge ? ResolveStrategy.MERGE : ResolveStrategy.REPLACE
+          resolve({ strategy, doForAllConflicts } as ResolveConflict)
+        },
+        onConfirm: () => {
+          this.hideModal()
+          resolve({ strategy: ResolveStrategy.KEEP_BOTH, doForAllConflicts } as ResolveConflict)
+        }
+      }
+      this.createModal(modal)
+    })
+  }
+
+  resolveDoCopyInsteadOfMoveForSpaces(): Promise<boolean> {
+    return new Promise<boolean>((resolve) => {
+      const modal = {
+        variation: 'danger',
+        icon: 'alarm-warning',
+        title: this.$gettext('Move not possible'),
```

Comment:
I'd prefer the wording "Copy here?" see https://github.com/owncloud/web/issues/6892#issuecomment-1245438400